velox/dwio/common/ScanSpec.cpp

```diff
@@ -277,15 +277,7 @@ bool testFilter(
     const TypePtr& type) {
   bool mayHaveNull = true;
 
-  // Has-null statistics is often not set. Hence, we supplement it with
-  // number-of-values statistic to detect no-null columns more often.
-  // Number-of-values is the number of non-null values. When it is equal to
-  // total number of values, we know there are no nulls.
   if (stats->getNumberOfValues().has_value()) {
-    if (stats->getNumberOfValues().value() == 0) {
```

Comment:
Can you investigate why the stats is about map keys instead of map rows themselves?  Is it a bug in Parquet statistics?

Reply:
The stats is retrieved from `RowGroupMetaDataPtr` and I think the parquet RowGroup only has stats for flat columns with no higher level structure. Its [definition](https://github.com/facebookincubator/velox/blob/main/velox/dwio/parquet/thrift/ParquetThriftTypes.h#L2923) can give some clue. @majetideepak maybe you have more insights about this?

Reply:
@Yuhta Even if we consider map rows, the `number-of-values(number of non-null values)` will be zero in this case right? And we always assign a `keys->addFilter(common::IsNotNull());` for key columns.
I wonder why a DWRF file with a map column with all nulls won't hit this.

Reply:
Because `getNumberOfValues` here should return the number of non null maps, not non-null keys

Reply:
@Yuhta Isn't the number of non-null maps here (1 row with a null map) equal to 0? Even if Parquet had this information, wouldn't we still end up here?

Reply:
> Because `getNumberOfValues` here should return the number of non null maps, not non-null map keys. If parquet does not have this information, we should return `nullopt` for the call on this column instead.

`getNumberOfValues` in parquet should return the number of key/value column in current case. Take the schema for example:
```
+--------+------------------+-------+
|col_name|data_type         |comment|
+--------+------------------+-------+
|i       |string            |null   |
|c       |map<string,string>|null   |
+--------+------------------+-------+
```
the `RowGroupMetaData` has 3 column chunks for column `i`, `map::key`, `map::value` and has no meta for `map`. This fast path actually doesn't apply the map rows as it's called only against keys or values, refer [code](https://github.com/facebookincubator/velox/blob/main/velox/dwio/parquet/reader/RepeatedColumnReader.cpp#L214-L220).

Reply:
@Yuhta Updated but note that this change means there is no row group level filter against map key/value anymore.

Reply:
That's expected, can you also delete `ListColumnReader::filterRowGroups`?  That one is also not correct